src/bootstrap/lib.rs

```diff
@@ -1167,7 +1167,14 @@ impl Build {
 
     /// Path to the python interpreter to use
     fn python(&self) -> &Path {
-        self.config.python.as_ref().unwrap()
+        if self.config.build.ends_with("apple-darwin") {
+            // Force /usr/bin/python3 on macOS for LLDB tests because we're loading the
+            // LLDB plugin's compiled module which only works with the system python
+            // (namely not Homebrew-installed python)
+            Path::new("/usr/bin/python3")
+        } else {
+            self.config.python.as_ref().unwrap()
```

Comment:
ah, one comment - this `unwrap` will fail is no python at all is installed. So it would be nice to give a more helpful error message, something like
```suggestion
            self.config.python.as_ref().expect("python is required for running LLDB or rustdoc tests")
```